### ISSUE TITLE
IDISP001 when using IServiceProvider.GetService<T>()

### ISSUE THREAD

**Issue by jeremyVignelles** [DotNetAnalyzers/IDisposableAnalyzers#293]
This code yields a IDISP001:
```cs
var securityProvider = services.GetService<Disposable>()
```

But this code doesn't
```cs
var securityProvider = services.GetRequiredService<Disposable>()
```

It should not warn, because the IServiceProvider is responsible to handle the object's lifetimes.

**Comment by JohanLarsson** [comment]
Thanks for reporting!
